PubMed.js

```diff
@@ -198,10 +198,16 @@ function detectWeb(doc, url) {
 
 	// determine if book or bookSection for PubMed Labs
 	var bookCitation = doc.getElementsByClassName('book-citation');
-	if (bookCitation.length > 0 && ZU.xpath(bookCitation, './/div[@class="affiliations"]')) {
+	if (bookCitation.length > 0 && ZU.xpath(doc, './/div[@class="affiliations"]')) {
 		// For a bookSection there are the affiliations of the authors of this
 		// section as well as the affiliations of the book authors.
-		var bookAffiliations = ZU.xpath(doc.getElementById('full-authors'), './/div[@class="affiliations"]/h3[@class="title"]').length > 1;
+		var bookAffiliations;
+		if (doc.getElementById('full-authors')) {
+			bookAffiliations = ZU.xpath(doc.getElementById('full-authors'), './/div[@class="affiliations"]/h3[@class="title"]').length > 1;
+		}
+		else {
+			bookAffiliations = ZU.xpath(doc, '//div[@id="full-view-heading"]//div[@class="affiliations"]//li').length > 1;
+		}
 		return bookAffiliations ? "bookSection" : "book";
```

Comment:
How about simplifying this into something like:
```js
// For a bookSection there is the affiliations section of the authors of this
// section as well as another affiliation sections for the book authors.
var isChapter = doc.querySelectorAll('full-view-heading div.affiliations, #full-view-heading div.affiliations).length > 1;
return isChapter ? "bookSection" : "book";
```
(untested code)? Shouldn't this do what we want?